libqtile/lazy.py

```diff
@@ -99,7 +99,7 @@ def when(
         focused: Match | None = None,
         if_no_focused: bool = False,
         layout: Iterable[str] | str | None = None,
-        when_floating: bool = True,
+        when_floating: bool | None = None,
```

Comment:
Can you keep the type narrowed to `bool` and just default to False here? or does that change semantics somehow?

Reply:
https://github.com/qtile/qtile/pull/3664#issue-1285547709

> when_floating is None by default
> when set to True, apply lazy call to floating windows only
> when set to False, apply lazy call to tiled windows only

Yes it changes, the current default (when_floating=True) is to match all the windows regardless if it is floating or not.

Reply:
I see, ok. Can you add all that to the commit message?

Reply:
Commit message expanded